src/Data/Text/Mixed/Rope.hs

```diff
@@ -567,17 +567,17 @@ utf16SplitAtPosition (Utf16.Position l c) rp = do
   Just (beforeLine <> beforeColumn, afterColumn)
 
 -- | Get a line by its 0-based index.
--- Returns @""@ if the index is out of bounds.
+-- Returns empty Rope if the index is out of bounds.
 -- The result doesn't contain @\\n@ characters.
 --
 -- >>> :set -XOverloadedStrings
 -- >>> map (\l -> getLine l "foo\nbar\n😊😊\n\n") [0..3]
 -- ["foo","bar","😊😊",""]
 --
 -- @since 0.3
-getLine :: Word -> Rope -> Text
+getLine :: Word -> Rope -> Rope
 getLine lineIdx rp =
-  case T.unsnoc firstLine of
+  fromText $ case T.unsnoc firstLine of
```

Comment:
If we want to return `Rope`, can we avoid `firstLineRope` and stick to `firstLineRope`? Otherwise we are doing a double conversion.